src/component/parser/json/converter/ProcessConverter.ts

```diff
@@ -87,6 +90,26 @@ export default class ProcessConverter {
     }
   }
 
+  private assignIncomingAndOutgoingIdsFromFlows(): void {
+    const fillShapeBpmnElementAttribute = (
+      shapeBpmnElementId: string,
+      shapeBpmnElementAttributeName: keyof Pick<ShapeBpmnElement, 'outgoingIds' | 'incomingIds'>,
```

Comment:
**thought:** for simplicity, we may declare instead:
```ts
shapeBpmnElementAttributeName: 'outgoingIds' | 'incomingIds'
```
The 2 attributes won't be renamed (as they are part of the public API) so there is no risk of error.
Anyway, let's keep this as it is now.

Reply:
Hi @tbouffard,

I understand your suggestion to simplify the code by declaring the property names directly.   
However, I personally prefer using `keyof Pick` to avoid potential naming errors in the future, especially using `replace` feature of IDE. This approach ensures that only the allowed property names are used, reducing the risk of introducing bugs or errors.

Thanks for sharing your thoughts though! :slightly\_smiling\_face:

Reply:
Okay, that's fine with me.